common/code/boost_utility.php

```diff
@@ -60,5 +60,13 @@ static function cmp_title($a,$b)
     { return strcmp($a['title'],$b['title']); }
 
     static function cmp_cxxstd($a,$b)
-    { return self::cmp(strcmp($a['cxxstd'],$b['cxxstd']),$a,$b); }
+    {
+        if (empty($a['cxxstd'])) {
+            $a['cxxstd'] = '9999';
+        }
+        if (empty($b['cxxstd'])) {
+            $b['cxxstd'] = '9999';
+        }
+        return self::cmp(strcmp($a['cxxstd'],$b['cxxstd']),$a,$b); 
+    }
```

Comment:
Could this whole function be simplified down to: `return self::cmp(intval($a['cxxstd']) <=> intval($b['cxxstd']), $a, $b);` ?

Reply:
Will check shortly.

Reply:
The spaceship operator (<=>) is only available since PHP v7. Also, `intval()` returns zero value for empty string or null values; then it will be the first place in the ascending order; that's what we expected to get.

So, what about to simplify it like this?
`return self::cmp(strcmp(self::empty_cxxstd_value($a), self::empty_cxxstd_value($b)),$a,$b);`

`empty_cxxstd_value` function will be like this:
```
    static function empty_cxxstd_value($a)
    { return empty($a['cxxstd']) ? '9999' : $a['cxxstd']; }
```

Reply:
In that case, let's not use arbitrary values like 9999. Instead write a compare function that takes two strings and treats empty ones as "greater" than non-empty ones. e.g.

```
if (empty(lhs)) {
   return empty(rhs) ? 0 : 1;
} else if (empty(rhs)) {
   return -1;
} else {
   return strcmp(lhs, rhs);
}
```